langchain4j/src/main/java/dev/langchain4j/service/output/ServiceOutputParser.java

```diff
@@ -18,6 +20,20 @@
 
 public class ServiceOutputParser {
 
+    /**
+     * JSON Pattern:<br />
+     * 
+     * <i>\\{(?:[^{}]|\\{(?:[^{}]|\\{(?:[^{}]|\\{[^{}]*\\})*\\})*\\})*\\}</i>: Matches JSON objects, accounting for nested objects.
+     * <i>|\\[(?:[^\\[\\]]|\\[(?:[^\\[\\]]|\\[(?:[^\\[\\]]|\\[[^\\[\\]]*\\])*\\])*\\])*\\]</i>: Matches JSON arrays, including arrays of objects.
+     */
+    private static final String JSON_PATTERN_REGEX = "\\{(?:[^{}]|\\{(?:[^{}]|\\{(?:[^{}]|\\{[^{}]*\\})*\\})*\\})*\\}" +
```

Comment:
Oh, that is a very complicated pattern... any chance to simplify it? or at least document it a bit more which part does what

Reply:
Ok, I see.

There is [another solution](https://github.com/langchain4j/langchain4j/pull/1605) to solve this issue which TBH I prefer more. It is also kind of a black box (at least for me), but time-tested in Groovy and more extendable if we'll want to forgive some LLM mistakes in terms of json syntax.

I hope you don't mind if we will proceed with #1605 and close this PR. Anyway, thank you so much for your efforts!

Reply:
No harm, we all share a common goal